packages/experimental/Menu/src/Menu/useMenu.ts

```diff
@@ -33,24 +40,38 @@ export const useMenu = (props: MenuProps): MenuState => {
   };
 };
 
-const useMenuOpenState = (isControlled: boolean, props: MenuProps): [boolean, (e: InteractionEvent, isOpen: boolean) => void] => {
+const useMenuOpenState = (
+  isControlled: boolean,
+  props: MenuProps,
+  parentSetOpen: (e: InteractionEvent, isOpen: boolean, bubble?: boolean) => void,
+): [boolean, (e: InteractionEvent, isOpen: boolean, bubble?: boolean) => void] => {
   const { defaultOpen, onOpenChange, open } = props;
   const initialState = typeof defaultOpen !== 'undefined' ? defaultOpen : !!open;
   const [openInternal, setOpenInternal] = React.useState<boolean>(initialState);
 
   const state = isControlled ? open : openInternal;
 
   const setOpen = React.useCallback(
-    (e: InteractionEvent, isOpen: boolean) => {
+    (e: InteractionEvent, isOpen: boolean, bubble?: boolean) => {
       const openPrev = state;
       if (!isControlled) {
-        setOpenInternal(isOpen);
+        if (!isOpen || lastCloseTimestamp + delayOpen <= Date.now()) {
+          setOpenInternal(isOpen);
+        }
+
+        if (!isOpen) {
+          lastCloseTimestamp = Date.now();
+        }
       }
       if (onOpenChange && openPrev !== isOpen) {
         onOpenChange(e, isOpen);
       }
+
+      if (bubble && parentSetOpen) {
+        parentSetOpen(e, isOpen, bubble);
+      }
```

Comment:
Post PR review: I wonder if we could have made an actual custom bubbling React Event, so that we can get the bubbling behavior for free? This is fine though

Reply:
I'm not sure what you're suggesting, are you talking about like adding a new useBubble hook?

Reply:
https://reactjs.org/docs/events.html

I'm suggesting something like creating an event named "onMenuDismiss" that is fired from the submenu, and bubbles up to the root menu. then we could have an event handler prop like 
```
onMenuDismiss={() => {
  if (rootMenu) { e.stopPropogation; }//stop it from bubbling if we're the top level menu.
  menu.dismisss(); // dismiss ourselves
  }
```